.pipelines/submodules-pipeline.yaml

```diff
@@ -244,32 +245,80 @@ stages:
     parameters:
       name: "cilium_e2e"
       displayName: Cilium
-      pipelineBuildImage: "$(BUILD_IMAGE)"
       testDropgz: true
-      clusterName: "submodules-ciliume2e"
+      clusterType: cilium-podsubnet-up
+      clusterName: "submodciliume2e"
+      vmSize: Standard_B2ms
+      k8sVersion: ""
+      windowsOsSku: ""
+      dependsOn: 'containerize'
 
   - template: singletenancy/cilium-overlay/cilium-overlay-e2e-job-template.yaml
     parameters:
       name: "cilium_overlay_e2e"
       displayName: Cilium on AKS Overlay
-      pipelineBuildImage: "$(BUILD_IMAGE)"
       testDropgz: true
-      clusterName: "submodules-overlaye2e"
+      clusterType: cilium-overlay-up
+      clusterName: "submodoverlaye2e"
+      vmSize: Standard_B2ms
+      k8sVersion: ""
+      windowsOsSku: ""
+      dependsOn: 'containerize'
 
   - template: singletenancy/aks-swift/e2e-job-template.yaml
     parameters:
       name: "aks_swift_e2e"
       displayName: AKS Swift
-      pipelineBuildImage: "$(BUILD_IMAGE)"
       testDropgz: true
+      clusterType: swift-byocni-up
       clusterName: "submodswift"
-      osSku: "Ubuntu"
+      vmSize: Standard_B2ms
+      k8sVersion: ""
+      windowsOsSku: ""
+      dependsOn: 'containerize'
+
+  - stage: delete
+    displayName: Delete Clusters
+    condition: always()
+    dependsOn:
+      - setup
+      - "aks_swift_e2e"
+      - "cilium_e2e"
+      - "cilium_overlay_cilium_e2e"
+    variables:
+      commitID: $[ stagedependencies.setup.env.outputs['EnvironmentalVariables.commitID'] ]
+    jobs:
```

Comment:
So if one of the stages fails, and we restart the cluster creation just for that step, this will run for every stage and will try to delete the already deleted cluster ?

Reply:
Yes.

Reply:
Will the delete fail if it tries deleting the already deleted cluster?

Reply:
No, the delete succeeds as the makefile exit code does not get read if set -e is not present.